### ISSUE TITLE
[Request]: Benny's Motorsport upgrade

### ISSUE THREAD

**Issue by tw1st31337** [Yimura/YimMenu#3360]
### Problem

The menu already adds all the slots for mods for a car with benny's upgrade already bought, so i thought that adding an option to add that upgrade would make car customization complete (plus buying the upgrade costs money :(( )

### Solution

adding benny's upgrade to the "slot" tab in "LS Customs" submenu

### Reason

would make car customization with the menu pretty much complete 

### Additional context

:D

**Comment by lonelybud** [comment]
bennys upgrade gives different model for a given vehicle. There names include `custom` at the end. So this will require checking if car is bennys, then finding out the model for that car & changing the model of your car somehow. Only solution I can see.

Right now you can do one thing - save the vehicle as persist vehicle and then change the model in the file saved on the disk.

**Comment by CypherMorgan** [comment]
Try spawning a benny's model of a specific car you want and max upgrade it or any mods u want, once done take the spawned car to ls customs or benny's autoshop and buy a loss and theft protection on that car. It will be saved in any of ur empty garages.

**Comment by tw1st31337** [comment]
yeah to be honest i didnt know that benny's upgraded cars had separate models with just "custom" (*or some number) in their name, but i'll keep this open if anyone ever wants to spend time on this

**Comment by CypherMorgan** [comment]
> yeah to be honest i didnt know that benny's upgraded cars had separate models with just "custom" in their name, but i'll keep this open if anyone ever wants to spend time on this

custom's only in the vehicle name but usually the spawn names/codes are like 'tenf' for normal and 'tenf2' for benny's (just an example).

**Comment by tw1st31337** [comment]
> > yeah to be honest i didnt know that benny's upgraded cars had separate models with just "custom" in their name, but i'll keep this open if anyone ever wants to spend time on this
> 
> custom's only in the vehicle name but usually the spawn names/codes are like 'tenf' for normal and 'tenf2' for benny's (just an example).

yeah my bad i was just about to edit my comment

**Comment by Tayipc38** [comment]
Ill see whether there is any effective way of detecting an upgradable vehicle and finding its upgraded counterpart.

**Comment by CypherMorgan** [comment]
> Ill see whether there is any effective way of detecting an upgradable vehicle and finding its upgraded counterpart.

If you manage to find a way, that might be used for other tunes too I assume, like the Drift Tunes from LSCM, or Arena War upgrades.

**Comment by Tayipc38** [comment]
> > Ill see whether there is any effective way of detecting an upgradable vehicle and finding its upgraded counterpart.
> 
> The game does this with scripts and just keeps a giant switch table of JOAATs that are upgradeable and their upgraded variant. We'd to copy the code like we do for clan logo.

Cant we somehow get a reference to the function housing the switch and use the function ourselves? I know we can patch scripts but never really heard "using" them.

**Comment by gir489returns** [comment]
> Cant we somehow get a reference to the function housing the switch and use the function ourselves? I know we can patch scripts but never really heard "using" them.

I asked MBG this before, if we could construct a script on the fly, call it and then get its result. However, from what he told me, it was either not possible, or too much effort when we could just reconstruct it easily in C++. If you can somehow jump into the script VM and have it jump back out, that'd be cool, but I don't know of a way, it's kind of a pseudo blackbox, since there's really only one way into the VM, and that's to have a script thread run.

**Comment by tw1st31337** [comment]
(also HEAVY nitpicking here but mod slots like "headlight covers" (in gauntlet classic custom for example) and the ability to put an ignition/remote bomb is missing. I'm guessing that the ability to make the car a personal vehicle and put insurance on it is missing for some good coding reason though)

edit: for some reason headlight covers in the gauntlet were under "ornaments" but if it works then it works

**Comment by TwystedMind** [comment]
> (also HEAVY nitpicking here but mod slots like "headlight covers" (in gauntlet classic custom for example) and the ability to put an ignition/remote bomb is missing. I'm guessing that the ability to make the car a personal vehicle and put insurance on it is missing for some good coding reason though)
> 
> 
> 
> edit: for some reason headlight covers in the gauntlet were under "ornaments" but if it works then it works

There is no longer an option to add Trackers or Insurance to vehicles under the LSC tab in YimMenu, this was removed intentionally a couple of months back since it just caused issues.

Just spawn the vehicle you want, add any customizations, then drive the vehicle into LS Customs and purchase the Tracker and Insurance to make it your Personal Vehicle.
